src/js/_enqueues/admin/site-icon.js

```diff
@@ -35,104 +60,177 @@
 			x1: x1,
 			y1: y1,
 			x2: xInit + x1,
-			y2: yInit + y1
+			y2: yInit + y1,
 		};
 
 		return imgSelectOptions;
 	}
 
-	$( function() {
-		// Build the choose from library frame.
-		$( '#choose-from-library-link' ).on( 'click', function() {
-			var $el = $(this);
-
-			// Create the media frame.
-			frame = wp.media({
-				button: {
-					// Set the text of the button.
-					text: $el.data('update'),
-					// Don't close, we might need to crop.
-					close: false
-				},
-				states: [
-					new wp.media.controller.Library({
-						title: $el.data( 'choose' ),
-						library: wp.media.query({ type: 'image' }),
-						date: false,
-						suggestedWidth: $el.data( 'size' ),
-						suggestedHeight: $el.data( 'size' )
-					}),
-					new wp.media.controller.SiteIconCropper({
-						control: {
-							params: {
-								width: $el.data( 'size' ),
-								height: $el.data( 'size' )
-							}
+	/**
+	 * Initializes the media frame for selecting or cropping an image.
+	 *
+	 * @since 6.5.0
+	 */
+	$chooseButton.on( 'click', function () {
+		var $el = $( this );
+
+		// Create the media frame.
+		frame = wp.media( {
+			button: {
+				// Set the text of the button.
+				text: $el.data( 'update' ),
+
+				// Don't close, we might need to crop.
+				close: false,
+			},
+			states: [
+				new wp.media.controller.Library( {
+					title: $el.data( 'choose-text' ),
+					library: wp.media.query( { type: 'image' } ),
+					date: false,
+					suggestedWidth: $el.data( 'size' ),
+					suggestedHeight: $el.data( 'size' ),
+				} ),
+				new wp.media.controller.SiteIconCropper( {
+					control: {
+						params: {
+							width: $el.data( 'size' ),
+							height: $el.data( 'size' ),
 						},
-						imgSelectOptions: calculateImageSelectOptions
-					})
-				]
-			});
-
-			frame.on( 'cropped', function( attachment) {
-				$( '#site_icon_hidden_field' ).val(attachment.id);
-				switchToUpdate(attachment.url);
+					},
+					imgSelectOptions: calculateImageSelectOptions,
+				} ),
+			],
+		} );
+
+		frame.on( 'cropped', function ( attachment ) {
+			$hiddenDataField.val( attachment.id );
+			switchToUpdate( attachment );
+			frame.close();
+
+			// Start over with a frame that is so fresh and so clean clean.
```

Comment:
This comment was flagged as odd, and I agree. Is this an Outkast reference or something?